src/filesys.cpp

```diff
@@ -740,5 +744,23 @@ bool Rename(const std::string &from, const std::string &to)
 	return rename(from.c_str(), to.c_str()) == 0;
 }
 
+bool bad_char(const char c)
+{
+	return !(isalnum(c) || c == '-' || c == '_' || c == ' ');
+}
+
+std::string SanitizeFilename(std::string input)
```

Comment:
this function seems to modify the input **and** return a new value.
either make it return an entirely new value (and take a `const`) or work on a `std::string*`